### DISCUSSION TITLE
Uso de Serviço

### DISCUSSION THREAD

**Discussion by RanieryCG** [balta-io/7196#16] [Perguntas e Respostas] [answered]
Seria errado em vez de usar Handler e Command, a API chamar um serviço do domínio?

Teríamos o padrão API -> Serviço -> repositório -> Serviço -> API.

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia @RanieryCG como vai?
Serviços de domínio ou domain services, são serviços compartilhados entre um ou mais domínios específicos.

Os handlers/commands são para gestão do fluxo de ações... pode ser que um handler chame vários serviços de domínio, mas chamar o serviço de domínio direto, não vejo como algo legal...

**Reply by RanieryCG** [reply]
@andrebaltieri, você tem algum livro ou vídeo para indicar sobre domain services? Pelo jeito não entendi bem o conceito.

**Reply by andrebaltieri** [reply]
Recomendo o Domain Driven Design do Eric Evans ou Implementing Domain Driven Design do Vaughan Vernon
